From 6aadaf4551aa84926e41831385400d8b117f92af Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:11:19 -0400
Subject: [ethernet] i40evf: don't shut down admin queue on error

Message-id: <1409663591-32531-37-git-send-email-sassmann@redhat.com>
Patchwork-id: 90512
O-Subject: [RHEL7.1 PATCH 036/148] i40evf: don't shut down admin queue on error
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

If the driver encounters an error while communicating with the PF
driver,  don't just shut down the admin queue unconditionally. The PF
may be delayed, and shutting down the admin queue causes it to fail
completely. If this happens, the VF will never complete initialization.

Change-ID: I6192e9d8caeefb738428c3597fa2f54fa400ce7f
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit c89a9e00f90cea4d824ff5890c86332a6687f830)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index d381bcc..4d547ab 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -2133,8 +2133,6 @@ err_alloc:
  kfree(adapter->vf_res);
  adapter->vf_res = NULL;
 err:
- if (hw->aq.asq.count)
-  i40evf_shutdown_adminq(hw); /* ignore error */
  /* Things went into the weeds, so try again later */
  if (++adapter->aq_wait_count > I40EVF_AQ_MAX_ERR) {
   dev_err(&pdev->dev, "Failed to communicate with PF; giving up.\n");
-- 
1.7.1